phpunit/wp-theme-json-test.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Tests theme.json related public APIs.
+ *
+ * @package Gutenberg
+ */
+
+class WP_Theme_Json_Test extends WP_UnitTestCase {
+
+	/**
+	 * Test that it reports correctly themes that have a theme.json.
+	 *
+	 * @group theme_json
+	 *
+	 * @covers wp_theme_has_theme_json
+	 */
+	public function test_theme_has_theme_json() {
+		switch_theme( 'block-theme' );
+		$this->assertTrue( wp_theme_has_theme_json() );
+	}
+
+	/**
+	 * Test that it reports correctly themes that do not have a theme.json.
+	 *
+	 * @group theme_json
+	 *
+	 * @covers wp_theme_has_theme_json
+	 */
+	public function test_theme_has_no_theme_json() {
+		switch_theme( 'default' );
+		$this->assertFalse( wp_theme_has_theme_json() );
+	}
+
+	/**
+	 * Test it reports correctly child themes that have a theme.json.
+	 *
+	 * @group theme_json
+	 *
+	 * @covers wp_theme_has_theme_json
+	 */
+	public function test_child_theme_has_theme_json() {
+		switch_theme( 'block-theme-child' );
+		$this->assertTrue( wp_theme_has_theme_json() );
+	}
+
+	/**
+	 * Test that it reports correctly child themes that do not have a theme.json
+	 * and the parent does.
+	 *
+	 * @group theme_json
+	 *
+	 * @covers wp_theme_has_theme_json
+	 */
+	public function test_child_theme_has_not_theme_json_but_parent_has() {
```

Comment:
There are a couple of test for which the template resolution does not work as expected, it reports being the same as the stylesheet.

- `block-theme-child` for `test_child_theme_has_theme_json` test. Resolved as expected:

```php
[26-Oct-2022 15:46:26 UTC] stylesheet /wordpress-phpunit/includes/../data/themedir1/block-theme-child
[26-Oct-2022 15:46:26 UTC] template   /wordpress-phpunit/includes/../data/themedir1/block-theme
```

- `block-theme-child-no-theme-json` for `test_child_theme_has_not_theme_json_but_parent_has`. Not resolved as expected and fails:

```php
[26-Oct-2022 15:45:47 UTC] stylesheet /var/www/html/wp-content/themes/block-theme-child-no-theme-json
[26-Oct-2022 15:45:47 UTC] template /var/www/html/wp-content/themes/block-theme-child-no-theme-json
```

- `default-child-no-theme-json` for `test_child_theme_has_not_theme_json_but_parent_has`. Not resolved as expected, but passes:

```php
[26-Oct-2022 15:47:04 UTC] stylesheet /var/www/html/wp-content/themes/default-child-no-theme-json
[26-Oct-2022 15:47:04 UTC] template   /var/www/html/wp-content/themes/default-child-no-theme-json
```

Reply:
Fixed by explicitely set up the theme directory https://github.com/WordPress/gutenberg/pull/45168/commits/92db7ba7fe63b2e1562e4ce6e492ff5edb662e87